pkg/deploy/generator/resources.go

```diff
@@ -1557,3 +1563,176 @@ func (g *generator) acrRbac() []*arm.Resource {
 		},
 	}
 }
+
+func (g *generator) devCIPool() *arm.Resource {
+	parts := []string{
+		fmt.Sprintf("base64ToString('%s')", base64.StdEncoding.EncodeToString([]byte("set -e\n\n"))),
+	}
+
+	for _, variable := range []string{
+		"azpToken",
+		"poolName",
+		"agentPrefix"} {
+		parts = append(parts,
+			fmt.Sprintf("'%s='''", strings.ToUpper(variable)),
+			fmt.Sprintf("parameters('%s')", variable),
+			"'''\n'",
+		)
+	}
+
+	trailer := base64.StdEncoding.EncodeToString([]byte(`lvextend -r -L +32G /dev/rootvg/homelv
+yum -y update -x WALinuxAgent
+# install az cli
+sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+cat > /etc/yum.repos.d/azure-cli.repo <<'EOF'
+[azure-cli]
+name=Azure CLI
+baseurl=https://packages.microsoft.com/yumrepos/azure-cli
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.microsoft.com/keys/microsoft.asc
+EOF
+
+# install common tooling
+yum -y install docker azure-cli jq git python36 python3-pip jq gcc \
+gpgme-devel glib2-devel make libseccomp libseccomp-devel glibc-static libassuan libassuan-devel
+
+pip3 install virtualenv
+
+# install golang
+GO_VERSION=1.13
+curl -sL https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz -o /tmp/go${GO_VERSION}.linux-amd64.tar.gz
+mkdir -p /usr/local/go
+tar -C /usr/local/go -xzf /tmp/go${GO_VERSION}.linux-amd64.tar.gz --strip-components=1 go
+ln -s /usr/local/go/bin/* /usr/bin/
+echo "GOROOT=/usr/local/go" | tee -a /etc/environment
+
+# Install azp agent
+wget https://vstsagentpackage.azureedge.net/agent/2.172.2/vsts-agent-linux-x64-2.172.2.tar.gz -O /tmp/vsts-agent-linux-x64-2.172.2.tar.gz
```

Comment:
Should this be templated so that we can update one variable to update the VSTS agent? They've also updated it, current version is now v2.173.0.